CRM/Admin/Form/SettingTrait.php

```diff
@@ -163,6 +163,7 @@ protected function addFieldsDefinedInSettingsMetadata() {
       $quickFormType = $this->getQuickFormType($props);
       if (isset($quickFormType)) {
         $options = CRM_Utils_Array::value('options', $props);
+        $required = CRM_Utils_Array::value('is_required', $props, FALSE);
```

Comment:
Add support for the `is_required` flag. Currently only applied when using `$this->add('type')`.

Reply:
```suggestion
        $required = $props['is_required'] ?? FALSE;
```

Reply:
That line belongs in another PR!